src/mono/mono/mini/aot-compiler.c

```diff
@@ -14814,10 +15207,13 @@ mono_aot_assemblies (MonoAssembly **assemblies, int nassemblies, guint32 jit_opt
 		res = aot_assembly (assemblies [i], jit_opts, &aot_opts);
 		if (res != 0) {
 			fprintf (stderr, "AOT of image %s failed.\n", assemblies [i]->image->name);
-			return 1;
+			goto early_exit;
```

Comment:
needs res = 1 to fix build errors with uninitialized variable.

Reply:
Ah, I didn't realize it wouldn't evaluate that aot_assembly only returns 0 or 1 from its code paths. Thanks!